client/my-sites/domains/domain-management/components/icann-verification/icann-verification-card.jsx

```diff
@@ -2,61 +2,164 @@
  * External dependencies
  */
 import React from 'react';
+import Gridicon from 'gridicons';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import classNames from 'classnames';
 
 /**
  * Internal dependencies
  */
-import IcannVerification from './icann-verification';
 import support from 'lib/url/support';
+import Button from 'components/button';
+import Card from 'components/card';
+import upgradesActions from 'lib/upgrades/actions';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import { domainManagementEditContactInfo } from 'my-sites/domains/paths';
+import { getRegistrantWhois } from 'state/selectors';
+import QueryWhois from 'components/data/query-whois';
 
-let IcannVerificationCard = React.createClass( {
-	propTypes: {
-		handleSubmit: React.PropTypes.func.isRequired,
-		submitting: React.PropTypes.bool.isRequired,
-		selectedDomainName: React.PropTypes.string.isRequired
-	},
+class IcannVerificationCard extends React.Component {
+	static propTypes = {
+		contactDetails: React.PropTypes.object,
+		explanationContext: React.PropTypes.string,
+		selectedDomainName: React.PropTypes.string.isRequired,
+		selectedSiteSlug: React.PropTypes.string.isRequired,
+	};
+
+	state = {
+		submitting: false,
+		emailSent: false,
+	};
+
+	componentWillUnmount() {
+		if ( this.timer ) {
+			clearTimeout( this.timer );
+			this.timer = null;
+		}
+	}
+
+	revertToWaitingState = () => {
+		this.timer = null;
+		this.setState( { emailSent: false } );
+	};
+
+	handleSubmit = ( event ) => {
+		event.preventDefault();
+
+		this.setState( { submitting: true } );
+
+		upgradesActions.resendIcannVerification( this.props.selectedDomainName, ( error ) => {
```

Comment:
I agree, but let's not do this in this PR. The discussion has been long enough and this refactor and subsequent testing would block this for another few days and I don't want to block Mike for that long. It might be a good idea for a mob programming session, though :)